<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN">

<!--Converted with LaTeX2HTML 2008 (1.71)
original version by:  Nikos Drakos, CBLU, University of Leeds
* revised and updated by:  Marcus Hennecke, Ross Moore, Herb Swan
* with significant contributions from:
  Jens Lippmann, Marek Rouchal, Martin Wilck and others -->
<HTML>
<HEAD>
<TITLE>HiC Contact Map Comaprison Using Graphlet Approach</TITLE>
<META NAME="description" CONTENT="HiC Contact Map Comaprison Using Graphlet Approach">
<META NAME="keywords" CONTENT="manuscript">
<META NAME="resource-type" CONTENT="document">
<META NAME="distribution" CONTENT="global">

<META HTTP-EQUIV="Content-Type" CONTENT="text/html; charset=utf-8">
<META NAME="Generator" CONTENT="LaTeX2HTML v2008">
<META HTTP-EQUIV="Content-Style-Type" CONTENT="text/css">

<LINK REL="STYLESHEET" HREF="manuscript.css">

</HEAD>

<BODY >

<P>

<P>

        
<P>
<H1 ALIGN=CENTER>HiC Contact Map Comaprison Using Graphlet Approach</H1>
<P ALIGN=CENTER><STRONG>
Behnam Rasoolian<SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="182" ALIGN="BOTTOM" BORDER="0"
 SRC="img1.png"
 ALT="$ ^{1,*}$"></SPAN>,
    Debswapna Bhattacharya<SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="182" ALIGN="BOTTOM" BORDER="0"
 SRC="img2.png"
 ALT="$ ^{1}$"></SPAN>
    <A NAME="tex2html1"
  HREF="#foot20"><SUP><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]"
 SRC="/usr/share/latex2html/icons/footnote.png"></SUP></A></STRONG>
<BR><FONT SIZE=-1>
<SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="182" ALIGN="BOTTOM" BORDER="0"
 SRC="img2.png"
 ALT="$ ^{1}$"></SPAN> Auburn University
        </FONT>
</P>
<HR>

<P>

<H3>Abstract:</H3>
<DIV CLASS="ABSTRACT">

<P>
In this study, we plan to find dissimilarities between normal cells and cancerous cells,
through investigating HiC contact maps. 
We suspect that there are systematic differences between how chromosomes are structured
between normal cells and cancerous cells.
we 
Ideally, it is desirable to compare 3D structures of 
cell in order to make such comparisons.
However, the main challenge that we face is that 
3D structure of a cell is not readily available. Based on
[<A
 HREF="manuscript.html#adhikari2016chromosome3d">1</A>], fluorescence in situ hybridizaiton
(FISH) is used for investigating 3D configuration of chromosomes.
However, this method can only be used locally and cannot map
the whole structure of the chromosomes.
In orther to find dissimilarities in the 3D structure of 
chromosomes, we used HiC dataset.
The HiC method, which was developed by [<A
 HREF="manuscript.html#lieberman2009comprehensive">2</A>], captures interactions between 
chromosomal fragments in kilobase resolution. Based on HiC data, an
<SPAN  CLASS="textit">interaction frequency (IF) </SPAN> matrix can be developed between <SPAN  CLASS="textit">loci</SPAN> at a desired resolution.
A cell <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="193" ALIGN="MIDDLE" BORDER="0"
 SRC="img3.png"
 ALT="$ IF_{ij}$"></SPAN> in an interaction frequency matrix captures the number of interaction detected
in HiC dataset between locus <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="179" ALIGN="BOTTOM" BORDER="0"
 SRC="img4.png"
 ALT="$ i$"></SPAN> and locus <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="193" ALIGN="MIDDLE" BORDER="0"
 SRC="img5.png"
 ALT="$ j$"></SPAN> in the genome.
An interaction matrix can be used to develop both inter- and intra-chromosomal interaction matrices.
We believe differences in interaction matrices can be found between normal cells and cancerous ones.
</DIV>
<P>
<H1><A NAME="SECTION00010000000000000000">
Introduction</A>
</H1>

<P>
Graphlet comparison is a novel method used to compare large networks in order to
find local similarities in them.
Authors of [<A
 HREF="manuscript.html#prvzulj2007biological">3</A>] provide a new measure of PPI
network comparison
based on 73 constraints. This is used in order to compare two large
networks in order to detect similarities.

<P>
[<A
 HREF="manuscript.html#milenkoviae2008uncovering">4</A>] 
 provide heuristics to compare two nodes based on some feature
(or signature) vectors, which is a 73-dimensional vector
<!-- MATH
 $\mathbf{s}^T
= [s_0, s_2, ..., s_{72}]$
 -->
<SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="200" ALIGN="MIDDLE" BORDER="0"
 SRC="img6.png"
 ALT="$ \mathbf{s}^T
= [s_0, s_2, ..., s_{72}]$"></SPAN> where <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="193" ALIGN="MIDDLE" BORDER="0"
 SRC="img7.png"
 ALT="$ s_i$"></SPAN> denotes the number of nodes in
the network that are part of an orbit <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="179" ALIGN="BOTTOM" BORDER="0"
 SRC="img4.png"
 ALT="$ i$"></SPAN>. 
<BR><SPAN  CLASS="textit">Important Result</SPAN>: Proteins with similar surroundings perform
similar functions.

<P>
In [<A
 HREF="manuscript.html#milenkovic2010cancer">5</A>], the same author investigates 
cancer-causing genes to find similarities in their signatures. After
clustering the genes based on <SPAN  CLASS="textit">signature similarity</SPAN> criteria,
some clusters contain a lot of cancerous genes.
They use 4 different clustering methods with varying parameters to cluster
the proteins. They then predict the cancer-relatedness of a protein 
<SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="179" ALIGN="BOTTOM" BORDER="0"
 SRC="img4.png"
 ALT="$ i$"></SPAN> using
an enrichment criteria <!-- MATH
 $\frac{k}{|C_i|}$
 -->
<SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="201" ALIGN="MIDDLE" BORDER="0"
 SRC="img8.png"
 ALT="$ \frac{k}{\vert C_i\vert}$"></SPAN> where <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="193" ALIGN="MIDDLE" BORDER="0"
 SRC="img9.png"
 ALT="$ C_i$"></SPAN> is the cluster
where protein <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="179" ALIGN="BOTTOM" BORDER="0"
 SRC="img4.png"
 ALT="$ i$"></SPAN> belongs and <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="179" ALIGN="BOTTOM" BORDER="0"
 SRC="img10.png"
 ALT="$ k$"></SPAN> is the number of cancer-causing
proteins in <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="193" ALIGN="MIDDLE" BORDER="0"
 SRC="img9.png"
 ALT="$ C_i$"></SPAN> and <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="195" ALIGN="MIDDLE" BORDER="0"
 SRC="img11.png"
 ALT="$ \vert C_i\vert$"></SPAN> is the size of <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="193" ALIGN="MIDDLE" BORDER="0"
 SRC="img9.png"
 ALT="$ C_i$"></SPAN>.

<P>
The authors of [<A
 HREF="manuscript.html#di2010fast">6</A>] generalized the idea of graphlets to 
ordered graphs were the nodes are labeled in ascending order.
As can be viewed, there are a total of 14 orbits for graphlets of size
2 and 3 since the label of graphlets is also included in toplogy.
In the new definition, <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="200" ALIGN="MIDDLE" BORDER="0"
 SRC="img12.png"
 ALT="$ d_v^i$"></SPAN> denotes the number of orbit <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="179" ALIGN="BOTTOM" BORDER="0"
 SRC="img4.png"
 ALT="$ i$"></SPAN> touches 
node <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="179" ALIGN="BOTTOM" BORDER="0"
 SRC="img13.png"
 ALT="$ v$"></SPAN>. Each node, is then assigned a vector of length 14 
<A NAME="tex2html2"
  HREF="#foot40"><SUP><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]"
 SRC="/usr/share/latex2html/icons/footnote.png"></SUP></A><!-- MATH
 $(d_v^1, d_v^2, ..., d_v^{14})$
 -->
<SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="199" ALIGN="MIDDLE" BORDER="0"
 SRC="img14.png"
 ALT="$ (d_v^1, d_v^2, ..., d_v^{14})$"></SPAN> 
and similarity of two nodes in two contact maps can be compared by
how geometrically close their corresponding vectors are.

<H1><A NAME="SECTION00020000000000000000">
Materials and Methods</A>
</H1>

<H3><A NAME="SECTION00020100000000000000">
Notations</A>
</H3>
In this paper, matrices and vectors are represented as bold
capital and bold small letters respectively.
matrix rows and columns are represented by a <SPAN  CLASS="textit">dot</SPAN>
notation. For example, the <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="179" ALIGN="BOTTOM" BORDER="0"
 SRC="img4.png"
 ALT="$ i$"></SPAN>th row of matrix <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="179" ALIGN="BOTTOM" BORDER="0"
 SRC="img15.png"
 ALT="$ M$"></SPAN> is
denoted by <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="193" ALIGN="MIDDLE" BORDER="0"
 SRC="img16.png"
 ALT="$ M_{i.}$"></SPAN> and its <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="193" ALIGN="MIDDLE" BORDER="0"
 SRC="img5.png"
 ALT="$ j$"></SPAN>th column is represented
by <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="193" ALIGN="MIDDLE" BORDER="0"
 SRC="img17.png"
 ALT="$ M_{.j}$"></SPAN>.

<P>
We denote the set of all contact maps in cell line <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="179" ALIGN="BOTTOM" BORDER="0"
 SRC="img18.png"
 ALT="$ T$"></SPAN> with 
<!-- MATH
 $\mathbb{C}^T$
 -->
<SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="182" ALIGN="BOTTOM" BORDER="0"
 SRC="img19.png"
 ALT="$ \mathbb{C}^T$"></SPAN>. If no particular cell line is addressed, the
subscripts are dropped.
Any arbitrary member of <!-- MATH
 $\mathbb{C}$
 -->
<SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="179" ALIGN="BOTTOM" BORDER="0"
 SRC="img20.png"
 ALT="$ \mathbb{C}$"></SPAN> is denoted by 
<SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="193" ALIGN="MIDDLE" BORDER="0"
 SRC="img21.png"
 ALT="$ C_{ij}$"></SPAN>, where <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="179" ALIGN="BOTTOM" BORDER="0"
 SRC="img4.png"
 ALT="$ i$"></SPAN> and <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="193" ALIGN="MIDDLE" BORDER="0"
 SRC="img5.png"
 ALT="$ j$"></SPAN> (<SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="193" ALIGN="MIDDLE" BORDER="0"
 SRC="img22.png"
 ALT="$ j \ge i$"></SPAN>) represent the two chromosomes involved. 
In human cells this set contains a total of 276 contact maps,
23 of which are intra-chromosomal and the rest are inter-chromosomal.
For ease of representations, intra-chromosomal contact maps are
distinguished by a single superscript, so we have <!-- MATH
 $C_{i,i} =
C_i$
 -->
<SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="193" ALIGN="MIDDLE" BORDER="0"
 SRC="img23.png"
 ALT="$ C_{i,i} =
C_i$"></SPAN>.

<P>
We denote the number of loci in a chromosome <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="179" ALIGN="BOTTOM" BORDER="0"
 SRC="img4.png"
 ALT="$ i$"></SPAN> by <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="193" ALIGN="MIDDLE" BORDER="0"
 SRC="img24.png"
 ALT="$ N_i$"></SPAN>.
The set of all loci involved in contact map <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="193" ALIGN="MIDDLE" BORDER="0"
 SRC="img21.png"
 ALT="$ C_{ij}$"></SPAN> is denoted 
by <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="193" ALIGN="MIDDLE" BORDER="0"
 SRC="img25.png"
 ALT="$ V_{ij}$"></SPAN>.
In intra-chromosomal contact maps, <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="193" ALIGN="MIDDLE" BORDER="0"
 SRC="img26.png"
 ALT="$ V_{i,i}$"></SPAN> containts only the 
loci of that particular chromosome (<!-- MATH
 $|V_i| = N_i$
 -->
<SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="195" ALIGN="MIDDLE" BORDER="0"
 SRC="img27.png"
 ALT="$ \vert V_i\vert = N_i$"></SPAN>), while in 
inter-chromosomal contact maps <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="193" ALIGN="MIDDLE" BORDER="0"
 SRC="img25.png"
 ALT="$ V_{ij}$"></SPAN> contains the loci in
the both of chromosomes involved (<!-- MATH
 $|V_{ij}| = N_i + N_j$
 -->
<SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="195" ALIGN="MIDDLE" BORDER="0"
 SRC="img28.png"
 ALT="$ \vert V_{ij}\vert = N_i + N_j$"></SPAN>).

<P>

<P>

<H2><A NAME="SECTION00021000000000000000">
Thresholding contact maps</A>
</H2>
In order to be able to extract graphlets, HiC contact maps should be modeled as
unweighted graphs where the nodes represent the loci and an edge between two 
nodes represent a <SPAN  CLASS="textit">significant</SPAN> interaction between the loci.
This can be achieved by thresholding the contact maps. The result
of the thresholding procedure is a binary matrix which also can serve as
an adjacency matrix for an unweighted, undirected graph. The graph can then be
used for orbit extraction.

<P>
When thresholding contact maps, it is necessary to make sure
that both global and local features are maintained. We could consider 
thresholding the contact maps by simply setting values above a fixed value to
one and the rest to zero; However, in practice, this method resulted in graphs
that capture the local structure of the contact maps poorly. This is because
intensities follow an exponential distribution with a mean close to zero
with a few very larges values that correspond to interactions along 
or close to the main diagonal of the contact maps.
Thus, picking relatively large numbers would result in ignoring interactions
that are far from the main diagonal while picking small values will lead to
capturing too many (insignificant) interactions.

<P>
In order to threshold the matrix so that both global and local patterns are
captured, we borrowed the concept of <SPAN  CLASS="textit">adaptive thresholding</SPAN> from image 
processing context. In this method, in order to be set, a pixel should have
an intensity larger than the average of non-zero intensities in its
<SPAN  CLASS="textit">neighborhood</SPAN>. The neighborhood is defined by a sliding kernel 
that passes through the contact map with the pixel at its middle at 
each step. Figure <A HREF="#local_thresholded_chr1_chr1"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]"
 SRC="/usr/share/latex2html/icons/crossref.png"></A> demonstrates result of 
this thresholding approach for intra-chromosomal contact maps of chromosome 1.
Refer to supplementary material for all 23 interchromosomal thresholding
results.

<DIV ALIGN="CENTER"><A NAME="local_thresholded_chr1_chr1"></A><A NAME="64"></A>
<TABLE>
<CAPTION ALIGN="BOTTOM"><STRONG>Figure:</STRONG>
Result of thresholding interchromosomal contact map of chromosome 1
    using a kernels of size <!-- MATH
 $5 \times 5$
 -->
<SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="193" ALIGN="MIDDLE" BORDER="0"
 SRC="img29.png"
 ALT="$ 5 \times 5$"></SPAN> for all cell lines. 
    The first row shows the thresholded
    maps. Second and third rows demonstrate pair-wise similarities and 
    differences between contact maps respectively.</CAPTION>
<TR><TD><DIV ALIGN="CENTER">

</DIV>    <IMG
  WIDTH="1080" HEIGHT="1440" ALIGN="BOTTOM" BORDER="0"
 SRC="./local_thresholded_chr1_chr1.png"
 ALT="Image local_thresholded_chr1_chr1"></TD></TR>
</TABLE>
</DIV>

<P>

<H2><A NAME="SECTION00022000000000000000">
Orbit Extraction</A>
</H2>
Once the thresholded contact maps are obtained, graphlets and orbits can be 
extracted. We used the <TT>orca</TT> package in <TT>R</TT> programming 
language to extract the graphlets. As a result of graphlet extraction, 
For each loci in each contact map, a <SPAN  CLASS="textit">signature vector</SPAN> of size
73 is created. Thus for each cell line, we would have 276 
<SPAN  CLASS="textit">signature matrices</SPAN> of
size <!-- MATH
 $|V^{ij}|\times 73$
 -->
<SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="200" ALIGN="MIDDLE" BORDER="0"
 SRC="img30.png"
 ALT="$ \vert V^{ij}\vert\times 73$"></SPAN>, where <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="182" ALIGN="BOTTOM" BORDER="0"
 SRC="img31.png"
 ALT="$ V^{ij}$"></SPAN> is the number of loci
involved in contact map between chromosomes <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="179" ALIGN="BOTTOM" BORDER="0"
 SRC="img4.png"
 ALT="$ i$"></SPAN> and <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="193" ALIGN="MIDDLE" BORDER="0"
 SRC="img5.png"
 ALT="$ j$"></SPAN>. Figure
<A HREF="#fig:graphlet_extraction"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]"
 SRC="/usr/share/latex2html/icons/crossref.png"></A> illustrates the process and results
of signature matrix extraction schematically.

<DIV ALIGN="CENTER"><A NAME="fig:graphlet_extraction"></A><A NAME="336"></A>
<TABLE>
<CAPTION ALIGN="BOTTOM"><STRONG>Figure:</STRONG>
Graphlet extraction for the four cell lines. For each
    loci in each contact map between chromosomes <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="179" ALIGN="BOTTOM" BORDER="0"
 SRC="img4.png"
 ALT="$ i$"></SPAN> and <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="193" ALIGN="MIDDLE" BORDER="0"
 SRC="img5.png"
 ALT="$ j$"></SPAN>, 
    the signature vectors of length 73 are extracted, resulting 
    in a <SPAN  CLASS="textit">signature matrix</SPAN> of size <!-- MATH
 $|V^{ij}| \times 73$
 -->
<SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="200" ALIGN="MIDDLE" BORDER="0"
 SRC="img30.png"
 ALT="$ \vert V^{ij}\vert\times 73$"></SPAN>,
    where <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="182" ALIGN="BOTTOM" BORDER="0"
 SRC="img31.png"
 ALT="$ V^{ij}$"></SPAN> 
    is the number of loci involved.</CAPTION>
<TR><TD><DIV ALIGN="CENTER">

</DIV>    <IMG
  WIDTH="1056" HEIGHT="816" ALIGN="BOTTOM" BORDER="0"
 SRC="./graphlet_extraction.png"
 ALT="Image graphlet_extraction"></TD></TR>
</TABLE>
</DIV>

<P>
For a particular
<!-- MATH
 $\mathbf{C}_{ij}$
 -->
<SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="193" ALIGN="MIDDLE" BORDER="0"
 SRC="img32.png"
 ALT="$ \mathbf{C}_{ij}$"></SPAN>, we denote <!-- MATH
 $\mathbf{S}_{ij}$
 -->
<SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="193" ALIGN="MIDDLE" BORDER="0"
 SRC="img33.png"
 ALT="$ \mathbf{S}_{ij}$"></SPAN> as its <SPAN  CLASS="textit">signature 
matrix</SPAN>. Each cell <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="193" ALIGN="MIDDLE" BORDER="0"
 SRC="img34.png"
 ALT="$ S_{ijlo}$"></SPAN> in <!-- MATH
 $\mathbf{S}_{ij}$
 -->
<SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="193" ALIGN="MIDDLE" BORDER="0"
 SRC="img33.png"
 ALT="$ \mathbf{S}_{ij}$"></SPAN> captures how many
times loci <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="179" ALIGN="BOTTOM" BORDER="0"
 SRC="img35.png"
 ALT="$ l$"></SPAN> in <!-- MATH
 $\mathbf{C}_{ij}$
 -->
<SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="193" ALIGN="MIDDLE" BORDER="0"
 SRC="img32.png"
 ALT="$ \mathbf{C}_{ij}$"></SPAN> occured as part of orbit <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="179" ALIGN="BOTTOM" BORDER="0"
 SRC="img36.png"
 ALT="$ o$"></SPAN>.

<P>

<DIV ALIGN="CENTER"><A NAME="graphlet_distance_schema"></A><A NAME="337"></A>
<TABLE>
<CAPTION ALIGN="BOTTOM"><STRONG>Figure:</STRONG>
Calculating pair-wise loci distances. For each loci (row) 
    in each
    contact map in MIT cell line, its distance is calculated based on
    equation <A HREF="#eq:distance_total"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]"
 SRC="/usr/share/latex2html/icons/crossref.png"></A> with the corresponding loci in
    leukemic cells. The result of this process is a 
    <SPAN  CLASS="textit">signature distance vector</SPAN> of size
    <!-- MATH
 $|V^{ij}| = N_i+N_j$
 -->
<SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="200" ALIGN="MIDDLE" BORDER="0"
 SRC="img37.png"
 ALT="$ \vert V^{ij}\vert = N_i+N_j$"></SPAN> for each contact map.
    </CAPTION>
<TR><TD><DIV ALIGN="CENTER">

</DIV>    <IMG
  WIDTH="834" HEIGHT="787" ALIGN="BOTTOM" BORDER="0"
 SRC="./graphlet_distance_schema.png"
 ALT="Image graphlet_distance_schema"></TD></TR>
</TABLE>
</DIV>

<DIV ALIGN="CENTER"><A NAME="graphlet_correlation_schema"></A><A NAME="338"></A>
<TABLE>
<CAPTION ALIGN="BOTTOM"><STRONG>Figure:</STRONG>
Calulating pair-wise orbit correlations. For each orbit (column)
    in each contact map in MIT cell line, its correlation with the
    same orbit in the same contact map 
    in leukemic cells is calculated. The result of
    this process is a <SPAN  CLASS="textit">signature correlation</SPAN> vector of size
    73 which captures how similar frequencies of two orbits are.
    In order to test our second hypothesis, we calculated averages
    across contact maps (along the vertical red arrow) to test 
    hypothesis <A HREF="#eq:h2a"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]"
 SRC="/usr/share/latex2html/icons/crossref.png"></A>
    and across orbits
    (along the horizontal red arrows) to test hypothesis <A HREF="#eq:h2b"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]"
 SRC="/usr/share/latex2html/icons/crossref.png"></A>.</CAPTION>
<TR><TD><DIV ALIGN="CENTER">

</DIV>    <IMG
  WIDTH="1056" HEIGHT="816" ALIGN="BOTTOM" BORDER="0"
 SRC="./graphlet_correlation_schema.png"
 ALT="Image graphlet_correlation_schema"></TD></TR>
</TABLE>
</DIV>

<P>
We divided the task of graphlet comparison into two parts: first we compare
graphlets from each contact map in normal cell lines (MIT) with the 
same contact maps in the other three Leukemic cells. Second we compare 
contact maps in a similar way but this time only between leukemic cells.
In the former case, the null hypothesis is that there is no difference
between contact maps of normal cells and leukemic cells and in the latter
case, the null hypotheis is that there is no difference between 
different leukemic cells.

<P>
We consider two measures of <SPAN  CLASS="textit">difference</SPAN> when comparing contact
map graphlets across cell lines. 
The first measure is <SPAN  CLASS="textit">signature distance vectors</SPAN> between
each contact map of two cell lines. 
Let  <!-- MATH
 $\mathbf{S}^A_{ij}$
 -->
<SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="200" ALIGN="MIDDLE" BORDER="0"
 SRC="img38.png"
 ALT="$ \mathbf{S}^A_{ij}$"></SPAN> and <!-- MATH
 $\mathbf{S}^B_{ij}$
 -->
<SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="200" ALIGN="MIDDLE" BORDER="0"
 SRC="img39.png"
 ALT="$ \mathbf{S}^B_{ij}$"></SPAN> be
signature matrices corresponding  to cells A and B
respectively, then we can caclulate the 
distance between each
loci (row) <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="179" ALIGN="BOTTOM" BORDER="0"
 SRC="img35.png"
 ALT="$ l$"></SPAN> in <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="182" ALIGN="BOTTOM" BORDER="0"
 SRC="img40.png"
 ALT="$ S^A$"></SPAN> and the the same loci in <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="182" ALIGN="BOTTOM" BORDER="0"
 SRC="img41.png"
 ALT="$ S^B$"></SPAN>
using the following formula from [<A
 HREF="manuscript.html#prvzulj2007biological">3</A>]:

<P>
<P></P>
<DIV ALIGN="CENTER" CLASS="mathdisplay"><A NAME="eq:distance_total"></A><!-- MATH
 \begin{equation}
D(\mathbf{S}^A_{ij}, \mathbf{S}^B_{ij}) =
    \frac{1}{73}\sqrt{\sum_{o=0}^{72}{\mathbf{D}_{.o}^2}}
\end{equation}
 -->
<TABLE CLASS="equation" CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="241" ALIGN="MIDDLE" BORDER="0"
 SRC="img42.png"
 ALT="$\displaystyle D(\mathbf{S}^A_{ij}, \mathbf{S}^B_{ij}) = \frac{1}{73}\sqrt{\sum_{o=0}^{72}{\mathbf{D}_{.o}^2}}$"></SPAN></TD>
<TD NOWRAP CLASS="eqno" WIDTH="10" ALIGN="RIGHT">
(<SPAN CLASS="eqn-number">1</SPAN>)</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P>
where elements of <!-- MATH
 $\mathbf{D}$
 -->
<SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="179" ALIGN="BOTTOM" BORDER="0"
 SRC="img43.png"
 ALT="$ \mathbf{D}$"></SPAN> can be calculated as follows:
<P></P>
<DIV ALIGN="CENTER" CLASS="mathdisplay"><A NAME="eq:distance_single"></A><!-- MATH
 \begin{equation}
D_{lo} = w_o \times
        \frac{log(S_{ijlo}^A+1) - log(S_{ijlo}^B+1)}
             {log(max(S_{ijlo}^A, S_{ijlo}^B) + 2)}
\end{equation}
 -->
<TABLE CLASS="equation" CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="229" ALIGN="MIDDLE" BORDER="0"
 SRC="img44.png"
 ALT="$\displaystyle D_{lo} = w_o \times \frac{log(S_{ijlo}^A+1) - log(S_{ijlo}^B+1)} {log(max(S_{ijlo}^A, S_{ijlo}^B) + 2)}$"></SPAN></TD>
<TD NOWRAP CLASS="eqno" WIDTH="10" ALIGN="RIGHT">
(<SPAN CLASS="eqn-number">2</SPAN>)</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P>

<P>
This process is illustrated in Figure <A HREF="#graphlet_distance_schema"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]"
 SRC="/usr/share/latex2html/icons/crossref.png"></A>.
Using this distance measure, we can quantify how two loci are close to
each other in terms of local neighborhood between the two contact maps.

<P>
The second measure of comparison that we use captures how 
similar two orbits are in terms of their count 
frequencies across loci between two contact maps. 
Each column in <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="193" ALIGN="MIDDLE" BORDER="0"
 SRC="img45.png"
 ALT="$ S_{ij}$"></SPAN> can provide information
regarding the <SPAN  CLASS="textit">frequency distribution</SPAN> of orbits throughout
the contact map <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="193" ALIGN="MIDDLE" BORDER="0"
 SRC="img21.png"
 ALT="$ C_{ij}$"></SPAN>. 
We can find how similar these distributions are to each other using
correlation measures.
These correlations can be calculated as the following:
<P></P>
<DIV ALIGN="CENTER" CLASS="mathdisplay"><A NAME="eq:correlation"></A><!-- MATH
 \begin{equation}
Corr(\mathbf{S}^A_{ij}, \mathbf{S}^B{ij}) = \mathbf{c}_{ij}
\end{equation}
 -->
<TABLE CLASS="equation" CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="202" ALIGN="MIDDLE" BORDER="0"
 SRC="img46.png"
 ALT="$\displaystyle Corr(\mathbf{S}^A_{ij}, \mathbf{S}^B{ij}) = \mathbf{c}_{ij}$"></SPAN></TD>
<TD NOWRAP CLASS="eqno" WIDTH="10" ALIGN="RIGHT">
(<SPAN CLASS="eqn-number">3</SPAN>)</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P>
where each element of <!-- MATH
 $\mathbf{c}$
 -->
<SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="179" ALIGN="BOTTOM" BORDER="0"
 SRC="img47.png"
 ALT="$ \mathbf{c}$"></SPAN> is calculated as below:
<P></P>
<DIV ALIGN="CENTER" CLASS="mathdisplay"><A NAME="eq:pearson"></A><!-- MATH
 \begin{equation}
c_{ijo} = Pearson(\mathbf{S}^A_{ij.o}, \mathbf{S}^B_{ij.o})
\end{equation}
 -->
<TABLE CLASS="equation" CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="202" ALIGN="MIDDLE" BORDER="0"
 SRC="img48.png"
 ALT="$\displaystyle c_{ijo} = Pearson(\mathbf{S}^A_{ij.o}, \mathbf{S}^B_{ij.o})$"></SPAN></TD>
<TD NOWRAP CLASS="eqno" WIDTH="10" ALIGN="RIGHT">
(<SPAN CLASS="eqn-number">4</SPAN>)</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P>
Alternatively, if we use MIC criterion we would have:
<P></P>
<DIV ALIGN="CENTER" CLASS="mathdisplay"><A NAME="eq:mic"></A><!-- MATH
 \begin{equation}
c_{ijo} = MIC(\mathbf{S}^A_{ij.o}, \mathbf{S}^B_{ij.o})
\end{equation}
 -->
<TABLE CLASS="equation" CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="202" ALIGN="MIDDLE" BORDER="0"
 SRC="img49.png"
 ALT="$\displaystyle c_{ijo} = MIC(\mathbf{S}^A_{ij.o}, \mathbf{S}^B_{ij.o})$"></SPAN></TD>
<TD NOWRAP CLASS="eqno" WIDTH="10" ALIGN="RIGHT">
(<SPAN CLASS="eqn-number">5</SPAN>)</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P>

<P>
For each contact map,  calculated
similarity between orbit distributions using Pearson's r 
correlation, which is computationally efficient.
However, pearson's r might not be able to capture
non-functional relationships between distributions. As a result, we
also used Maximal Information Coefficient (MIC) 
[<A
 HREF="manuscript.html#reshef2011detecting">7</A>] in order to compare
correlations. MIC calculates mutual information (MI) between two
distributions, but utilizes dynamic programming in order adjust
bin sizes and numbers in order to achieve highest MI.
MIC values between two variables fall between 0 and 1,
with 0 meaning the two variables are completely independent
and 1 meaning one is dependant on the other.
We used both Pearson's r and MI in order to compare orbit
frequencies. Although results from both approaches were more
or less consistent, MIC showed higher robustness than Pearson's 
r method.

<P>

<H2><A NAME="SECTION00023000000000000000">
Hypotheses</A>
</H2>

<P>
With regard to the two aforementioned comparison methods, we define
our two null hypotheses as the following:

<P>

<OL>
<LI>Average signature vector distance between each 
        contact map of MIT cells and
        their counterparts in the three leukemic cells
        are equal to <SPAN CLASS="MATH">0</SPAN>. In mathematical terms:
        <!-- MATH
 $\forall i,j \in \{1, 2 ..., 24\}$
 -->
<SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="195" ALIGN="MIDDLE" BORDER="0"
 SRC="img50.png"
 ALT="$ \forall i,j \in \{1, 2 ..., 24\}$"></SPAN>
        <P></P>
<DIV ALIGN="CENTER" CLASS="mathdisplay"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="287" ALIGN="BOTTOM" BORDER="0"
 SRC="img51.png"
 ALT="\begin{align*}\begin{split}H_0&amp;: \frac{1}{\vert V_{ij}\vert}\sum_{l=0}^{\vert V_...
... V_{ij}\vert}\sum_{l=0}^{\vert V_{ij}\vert}{d_{ijl}} &gt; 0 \end{split}\end{align*}"></SPAN></TD>
<TD NOWRAP CLASS="eqno" WIDTH="10" ALIGN="RIGHT">
<A NAME="eq:h1">(<SPAN CLASS="eqn-number">6</SPAN>)</A></TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P>

<P>
where 
        <!-- MATH
 $\mathbf{d}_{ij} = D(S^{MIT}_{ij}, S^L_{ij})$
 -->
<SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="200" ALIGN="MIDDLE" BORDER="0"
 SRC="img52.png"
 ALT="$ \mathbf{d}_{ij} = D(S^{MIT}_{ij}, S^L_{ij})$"></SPAN> and <!-- MATH
 $L \in \{ALL, RL, CALL4\}$
 -->
<SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="195" ALIGN="MIDDLE" BORDER="0"
 SRC="img53.png"
 ALT="$ L \in \{ALL, RL, CALL4\}$"></SPAN>.

<P>
</LI>
<LI><OL>
<LI>Average orbit correlations within a contact map of MIT cell, 
                    and its counterpart in leukemic cells is equal to 1
                    for each contact map.
                Again in mathematical terms:
                <!-- MATH
 $\forall i,j \in \{1, 2 ..., 24\}$
 -->
<SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="195" ALIGN="MIDDLE" BORDER="0"
 SRC="img50.png"
 ALT="$ \forall i,j \in \{1, 2 ..., 24\}$"></SPAN>
                <P></P>
<DIV ALIGN="CENTER" CLASS="mathdisplay"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="279" ALIGN="BOTTOM" BORDER="0"
 SRC="img54.png"
 ALT="\begin{align*}\begin{split}H_0&amp;: \frac{1}{73}\sum_{o=0}^{72}{c_{ijo}} = 1 \\ H_1&amp;: \frac{1}{73}\sum_{o=0}^{72}{c_{ijo}} &lt; 1 \end{split}\end{align*}"></SPAN></TD>
<TD NOWRAP CLASS="eqno" WIDTH="10" ALIGN="RIGHT">
<A NAME="eq:h2b">(<SPAN CLASS="eqn-number">7</SPAN>)</A></TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P>
where <!-- MATH
 $\mathbf{c}_{ij} = Corr(S^{MIT}_{ij}, S^L_{ij})$
 -->
<SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="200" ALIGN="MIDDLE" BORDER="0"
 SRC="img55.png"
 ALT="$ \mathbf{c}_{ij} = Corr(S^{MIT}_{ij}, S^L_{ij})$"></SPAN>, <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="179" ALIGN="BOTTOM" BORDER="0"
 SRC="img56.png"
 ALT="$ L$"></SPAN> 
                        is the same as above and <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="179" ALIGN="BOTTOM" BORDER="0"
 SRC="img57.png"
 ALT="$ N$"></SPAN> 
                        is the total number of contact maps which 
                        is equal to <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="179" ALIGN="BOTTOM" BORDER="0"
 SRC="img58.png"
 ALT="$ 275$"></SPAN> in our case.

<P>
</LI>
<LI>Average orbit correlations  of MIT cell
                    and leukemic cells across contact maps is equal to 1
                    for each orbit.
                Again in mathematical terms:

<P>

<P></P>

<P>
<!-- MATH
 $\forall o \in \{0, 1 ... 72\}$
 -->
<SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="195" ALIGN="MIDDLE" BORDER="0"
 SRC="img59.png"
 ALT="$ \forall o \in \{0, 1 ... 72\}$"></SPAN>
                <P></P>
<DIV ALIGN="CENTER" CLASS="mathdisplay"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="284" ALIGN="BOTTOM" BORDER="0"
 SRC="img60.png"
 ALT="\begin{align*}\begin{split}H_0&amp;: \frac{1}{N}\sum_{i=1}^{23}{\sum_{j=i}^{23}{c_{i...
...\frac{1}{N}\sum_{i=1}^{23}{\sum_{j=i}^{23}{c_{ijo}}} &lt; 1 \end{split}\end{align*}"></SPAN></TD>
<TD NOWRAP CLASS="eqno" WIDTH="10" ALIGN="RIGHT">
<A NAME="eq:h2a">(<SPAN CLASS="eqn-number">8</SPAN>)</A></TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P>
                        <SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="179" ALIGN="BOTTOM" BORDER="0"
 SRC="img56.png"
 ALT="$ L$"></SPAN> and <!-- MATH
 $\mathbf{c}_{ij}$
 -->
<SPAN CLASS="MATH"><IMG
 WIDTH="1069" HEIGHT="193" ALIGN="MIDDLE" BORDER="0"
 SRC="img61.png"
 ALT="$ \mathbf{c}_{ij}$"></SPAN> are the same as above.

<P>
</LI>
</OL>
</LI>
</OL>
<H1><A NAME="SECTION00030000000000000000">
results</A>
</H1>

<P>

<H2><A NAME="SECTION00031000000000000000">
Hypothesis 1: Orbit Distance Comparison</A>
</H2>

<DIV ALIGN="CENTER"><A NAME="fig:orbit-distances_all"></A><A NAME="fig:orbit-distances_intra"></A><A NAME="fig:orbit-distances"></A><A NAME="247"></A>
<TABLE>
<CAPTION ALIGN="BOTTOM"><STRONG>Figure:</STRONG>
   Pair-wise graphlet signature difference for
                (a) all 276 contact maps.
                (b) only the 23 intra-chromosomal contact maps.
                Each point on a graph is the result of averaging all
                the distances acroos all loci of that contact map.
                As can be viewd, although inter-chromosomal graphlets 
                do not show significant differences, RL's inter-chromosomal
                graphlets seem to be closer to MIT's. Also in some contact
                maps ALL graphlets show greater distance from MIT graphlets
                than the other two cell lines.
             </CAPTION>
<TR><TD><DIV ALIGN="CENTER">

</DIV>    <BR>
<IMG
 WIDTH="1069" HEIGHT="172" BORDER="0"
 SRC="img62.png"
 ALT="\begin{subfigure}
% latex2html id marker 235
[b]{\textwidth}
\includegraphics[width=\textwidth]{figures/orbit-distances_all.png}
\caption{}
\end{subfigure}">
<BR>

<P>
<BR>
<IMG
 WIDTH="1069" HEIGHT="92" BORDER="0"
 SRC="img63.png"
 ALT="\begin{subfigure}
% latex2html id marker 241
[b]{\textwidth}
\includegraphics[width=\textwidth]{figures/orbit-distances_intra.png}
\caption{}
\end{subfigure}"></TD></TR>
</TABLE>
</DIV>

Result of pair-wise contact map graphlet distances can is
demonstrated in Figure <A HREF="#fig:orbit-distances"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]"
 SRC="/usr/share/latex2html/icons/crossref.png"></A>.
Each point on the graph is the
average distance between signature vector of two same loci 
in the two cell lines as in Equation <A HREF="#eq:h1"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]"
 SRC="/usr/share/latex2html/icons/crossref.png"></A>.
in order to test hypothesis <A HREF="#eq:h1"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]"
 SRC="/usr/share/latex2html/icons/crossref.png"></A> a one sided t-test
was conducted. The resulting p-values showed highly significant differeces
between graphlets extracted from all contact maps. Refer to supplementary
material for results of all hypotheis tests.

<P>

<H2><A NAME="SECTION00032000000000000000">
Hypothesis 2: Orbit distribution comparison</A>
</H2>
We caclulated pair-wise Pearson's r correlation for each orbit in each of the 276
contact maps from MIT data and ALL, RL and CALL4 data separately. 
Figure <A HREF="#fig:contact_maps_correlations"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]"
 SRC="/usr/share/latex2html/icons/crossref.png"></A>
shows average orbit correlations across all contact maps
(average along the red <SPAN  CLASS="textit">vertical</SPAN> arrow in figure 
<A HREF="#graphlet_correlation_schema"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]"
 SRC="/usr/share/latex2html/icons/crossref.png"></A>), while
figure <A HREF="#fig:orbits_correlations"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]"
 SRC="/usr/share/latex2html/icons/crossref.png"></A> 
demonstrates average correlations across all 72 orbits within each  contact map
(average along the red <SPAN  CLASS="textit">horizontal</SPAN> arrows in figure 
<A HREF="#graphlet_correlation_schema"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]"
 SRC="/usr/share/latex2html/icons/crossref.png"></A>).
It is worth mentioning that <SPAN  CLASS="textit">interchromosomal</SPAN> thresholded contact maps 
represent
a biparticde graph with the loci from each chromosome on one side. Due to this
bipartide nature of the graphs in inter-chromosomal maps,
count of certain orbits is always 0, resulting in
a correlation values of 0 for them as well.
We ignored these values  when we calculated averages across orbits 
in figure <A HREF="#fig:contact_maps_correlations_all"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]"
 SRC="/usr/share/latex2html/icons/crossref.png"></A> since they
would result in a bias towards zero in averages. You can see the bias in 
figure <A HREF="#fig:orbits_correlations_all"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]"
 SRC="/usr/share/latex2html/icons/crossref.png"></A> where average correlations of obits
3, 9, 10-14, 20-34, 39-48, and 51-72 are close to zero. In fact all correlations
corresponding to these orbits are 0 except for the ones between the same 
chromosomes which are illustrated in Figure 
<A HREF="#fig:orbits_correlations_intra"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]"
 SRC="/usr/share/latex2html/icons/crossref.png"></A>.

<P>

<DIV ALIGN="CENTER"><A NAME="fig:contact_maps_correlations_all"></A><A NAME="fig:contact_maps_correlations_intra"></A><A NAME="fig:contact_maps_correlations"></A><A NAME="277"></A>
<TABLE>
<CAPTION ALIGN="BOTTOM"><STRONG>Figure:</STRONG>
pair-wise average contact map orbit correlations
             (a) for all contact maps
             (b) only for intra-chromosomal contact maps
             These values are calculated by averaging over 
             pairwise correlations of all orbits in a 
             contact map.
             </CAPTION>
<TR><TD><DIV ALIGN="CENTER">

</DIV>    <BR>
<IMG
 WIDTH="1069" HEIGHT="172" BORDER="0"
 SRC="img64.png"
 ALT="\begin{subfigure}
% latex2html id marker 265
[b]{\textwidth}
\includegraphics[w...
...xtwidth]{figures/contact_maps_correlations_all.png}
\caption{}
\end{subfigure}">
<BR>
    <BR>
<IMG
 WIDTH="1069" HEIGHT="92" BORDER="0"
 SRC="img65.png"
 ALT="\begin{subfigure}
% latex2html id marker 271
[b]{\textwidth}
\includegraphics[w...
...width]{figures/contact_maps_correlations_intra.png}
\caption{}
\end{subfigure}"></TD></TR>
</TABLE>
</DIV>

<P>

<DIV ALIGN="CENTER"><A NAME="fig:orbits_correlations_all"></A><A NAME="fig:orbits_correlations_intra"></A><A NAME="fig:orbits_correlations"></A><A NAME="352"></A>
<TABLE>
<CAPTION ALIGN="BOTTOM"><STRONG>Figure:</STRONG>
pair-wise average orbit correlations.
             In figure <A HREF="#fig:orbits_correlations_all"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]"
 SRC="/usr/share/latex2html/icons/crossref.png"></A>, each point
             in the graph is the result of averaging pair-wise
             orbit correlations over all contact maps, while
             each point in figure <A HREF="#fig:orbits_correlations_intra"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]"
 SRC="/usr/share/latex2html/icons/crossref.png"></A>
             are averaged only over intra-chromosomal contact maps.
             Counts for certain orbits are always zero in inter-chromosomal
             maps, leading to average value close to zero in 
             <A HREF="#fig:orbits_correlations_all"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]"
 SRC="/usr/share/latex2html/icons/crossref.png"></A>.
             </CAPTION>
<TR><TD><DIV ALIGN="CENTER">

</DIV>    <BR>
<IMG
 WIDTH="1069" HEIGHT="172" BORDER="0"
 SRC="img66.png"
 ALT="\begin{subfigure}
% latex2html id marker 281
[b]{\textwidth}
\includegraphics[w...
...2\paperheight]{figures/orbits_correlations_all.png}
\caption{}
\end{subfigure}">
<BR>
    <BR>
<IMG
 WIDTH="1069" HEIGHT="92" BORDER="0"
 SRC="img67.png"
 ALT="\begin{subfigure}
% latex2html id marker 287
[b]{\textwidth}
\includegraphics[w...
...paperheight]{figures/orbits_correlations_intra.png}
\caption{}
\end{subfigure}"></TD></TR>
</TABLE>
</DIV>

We have conducted one-sided
t-test in order to test whether the average correlations across
contact maps is equal to 1 (Equation <A HREF="#eq:h2a"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]"
 SRC="/usr/share/latex2html/icons/crossref.png"></A>) and whether the average 
correlations across orbits is equal to 1 (Equation <A HREF="#eq:h2b"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]"
 SRC="/usr/share/latex2html/icons/crossref.png"></A>). The results
for both test showed that all values are significantly less than 1.
Please refer to supplementary material for result of the t-tests.

<H2><A NAME="SECTION00033000000000000000">
Comparing Cancer cells</A>
</H2>

<DIV ALIGN="CENTER"><A NAME="fig:orbit_distances_intra_cancer"></A><A NAME="fig:orbits_correlations_intra_cancer"></A><A NAME="fig:orbits_cancer_intra"></A><A NAME="314"></A>
<TABLE>
<CAPTION ALIGN="BOTTOM"><STRONG>Figure:</STRONG>
</CAPTION>
<TR><TD><DIV ALIGN="CENTER">

</DIV>    <BR>
<IMG
 WIDTH="1069" HEIGHT="172" BORDER="0"
 SRC="img68.png"
 ALT="\begin{subfigure}
% latex2html id marker 302
[b]{\textwidth}
\includegraphics[w...
...extwidth]{figures/orbit-distances_intra_cancer.png}
\caption{}
\end{subfigure}">
<BR>
    <BR>
<IMG
 WIDTH="1069" HEIGHT="92" BORDER="0"
 SRC="img69.png"
 ALT="\begin{subfigure}
% latex2html id marker 308
[b]{\textwidth}
\includegraphics[w...
...idth]{figures/orbits_correlations_intra_cancer.png}
\caption{}
\end{subfigure}"></TD></TR>
</TABLE>
</DIV>

<P>
We also conducted the same comparison with the same hypothesis among cancer cells.
The results for intra-chromosomal distances and correlations can be found in
figure <A HREF="#fig:orbits_cancer_intra"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]"
 SRC="/usr/share/latex2html/icons/crossref.png"></A>. As can be noticed, CALL4 and RL cell
line exhibit a significantly higher average correlations for the majority of
the chromosomes (figure <A HREF="#fig:orbit_distances_intra_cancer"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]"
 SRC="/usr/share/latex2html/icons/crossref.png"></A>) and almost in
all orbits (figure <A HREF="#fig:orbits_correlations_intra_cancer"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]"
 SRC="/usr/share/latex2html/icons/crossref.png"></A>).

<H1><A NAME="SECTION00040000000000000000">
discussion</A>
</H1>

<H1><A NAME="SECTION00050000000000000000">
Resources</A>
</H1>
<SPAN  CLASS="textbf">Hi-C Datasets:</SPAN>

<OL>
<LI>https://github.com/rasoolianbehnam/watsonCode base for this article
</LI>
<LI>http://sysbio.rnet.missouri.edu/T0510/tmp_download/link_to_download_genome_data/
        Datasets including cancerous cells
</LI>
<LI>https://bcm.app.box.com/v/aidenlab/folder/11234760671Original Datasets
</LI>
</OL>

<P>

<H2><A NAME="SECTION00060000000000000000">
Bibliography</A>
</H2><DL COMPACT><DD><P></P><DT><A NAME="adhikari2016chromosome3d">1</A>
<DD>
Badri Adhikari, Tuan Trieu, and Jianlin Cheng.
<BR>Chromosome3d: reconstructing three-dimensional chromosomal structures
  from hi-c interaction frequency data using distance geometry simulated
  annealing.
<BR><EM>BMC genomics</EM>, 17(1):886, 2016.

<P></P><DT><A NAME="lieberman2009comprehensive">2</A>
<DD>
Erez Lieberman-Aiden, Nynke&nbsp;L Van&nbsp;Berkum, Louise Williams, Maxim Imakaev,
  Tobias Ragoczy, Agnes Telling, Ido Amit, Bryan&nbsp;R Lajoie, Peter&nbsp;J Sabo,
  Michael&nbsp;O Dorschner, et&nbsp;al.
<BR>Comprehensive mapping of long-range interactions reveals folding
  principles of the human genome.
<BR><EM>science</EM>, 326(5950):289-293, 2009.

<P></P><DT><A NAME="prvzulj2007biological">3</A>
<DD>
Nata&#353;a Pr&#382;ulj.
<BR>Biological network comparison using graphlet degree distribution.
<BR><EM>Bioinformatics</EM>, 23(2):e177-e183, 2007.

<P></P><DT><A NAME="milenkoviae2008uncovering">4</A>
<DD>
Tijana Milenkovi&#230; and Nata&#353;a Pr&#382;ulj.
<BR>Uncovering biological network function via graphlet degree
  signatures.
<BR><EM>Cancer informatics</EM>, 6:257, 2008.

<P></P><DT><A NAME="milenkovic2010cancer">5</A>
<DD>
Tijana Milenkovi&#263;, Vesna Memi&#353;evi&#263;, Anand&nbsp;K Ganesan, and
  Nata&#353;a Pr&#382;ulj.
<BR>Systems-level cancer gene identification from protein interaction
  network topology applied to melanogenesis-related functional genomics data.
<BR><EM>Journal of the Royal Society Interface</EM>, 7(44):423-437, 2010.

<P></P><DT><A NAME="di2010fast">6</A>
<DD>
Pietro Di&nbsp;Lena, Piero Fariselli, Luciano Margara, Marco Vassura, and Rita
  Casadio.
<BR>Fast overlapping of protein contact maps by alignment of
  eigenvectors.
<BR><EM>Bioinformatics</EM>, 26(18):2250-2258, 2010.

<P></P><DT><A NAME="reshef2011detecting">7</A>
<DD>
David&nbsp;N Reshef, Yakir&nbsp;A Reshef, Hilary&nbsp;K Finucane, Sharon&nbsp;R Grossman, Gilean
  McVean, Peter&nbsp;J Turnbaugh, Eric&nbsp;S Lander, Michael Mitzenmacher, and Pardis&nbsp;C
  Sabeti.
<BR>Detecting novel associations in large data sets.
<BR><EM>science</EM>, 334(6062):1518-1524, 2011.
</DL>

<BR><HR><H4>Footnotes</H4>
<DL>
<DT><A NAME="foot20">...#tex2html_wrap_inline699#</A><A
 HREF="manuscript.html#tex2html1"><SUP><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]"
 SRC="/usr/share/latex2html/icons/footnote.png"></SUP></A></DT>
<DD>
    Tel: +1 334 5212814; Email: bzr0014@auburn.edu

</DD>
<DT><A NAME="foot40">... 14</A><A
 HREF="manuscript.html#tex2html2"><SUP><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]"
 SRC="/usr/share/latex2html/icons/footnote.png"></SUP></A></DT>
<DD>number of orbits in graphlets of size 2 and 3

</DD>
</DL><BR><HR>

</BODY>
</HTML>
